dali/operators/decoder/audio/utils.h

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_DECODER_AUDIO_UTILS_H_
+#define DALI_OPERATORS_DECODER_AUDIO_UTILS_H_
+
+#include <functional>
+#include <numeric>
+#include <utility>
+#include <vector>
+#include <cassert>
+#include <cmath>
+#include "dali/core/error_handling.h"
+#include "dali/core/span.h"
+
+namespace dali {
+
+/**
+ * Downmixing data to mono-channel. It's safe to do it in-place.
+ *
+ * Assumes, that `out` buffer is properly allocated
+ * (i.e. has `n_in`/`weights.size()` elements at least)
+ *
+ * @param weights Specifies weight for every channel.
+ */
+template<typename T>
+void Downmixing(T *out, const T *in, size_t n_in, const std::vector<int> &weights) {
```

Comment:
1. I'd much rather have a verb for a function name (i.e. `Downmix`).\
2. The weights _must_ be floating point.
3. I'd prefer to pass the weights as a span or even a pointer - and have an explicit parameter for the number of channels. Now it's hidden and the calls are hard to read.
4. I'd much rather have n_out (or, better, num_samples) that is the number of samples, not samples*channels.